performance-best-practice.md

```diff
@@ -0,0 +1,447 @@
+---
+title: Optimizing TiDB Settings for Maximum Performance
+summary: Learn how to optimize TiDB performance by configuring key settings and addressing edge cases
+---
+
+# Optimizing TiDB Settings for Maximum Performance
+
+This guide provides essential information on optimizing TiDB for maximum performance. We'll explore:
+- Performance best practices for common workloads
+- Strategies for addressing challenging performance scenarios
+
+> **Important Note:**
+>
+> These optimization techniques are particularly valuable when pursuing extreme performance in TiDB. However, it's important to note that performance tuning often involves trade-offs between multiple factors, and there is no single "silver bullet" solution. Some of the techniques described here may use experimental features, which are specifically called out. While these optimizations can yield significant performance gains, they may not be suitable for stable production environments and should be used with caution.
+
+
+# Why This Matters
+
+Optimizing TiDB for peak performance requires careful tuning of various settings. In many cases, achieving optimal performance involves adjusting configurations beyond their default values.
+
+While default settings are chosen for stability, maximizing performance often requires more aggressive configurations and sometimes the use of experimental features. This approach is based on insights gained from numerous real-world deployments and performance optimization efforts.
+
+This guide explains these non-default settings, detailing their benefits and potential trade-offs. It's designed to help you make informed decisions when fine-tuning TiDB for your specific workload requirements.
+
+# Key Settings for Common Workloads
+
+The following suggested settings cover the most common optimizations for improving TiDB performance:
+
+- Enhancing query plan caching
+- Optimizing the query optimizer's behavior
+- More aggressive use of TiKV's Titan storage engine
+
+These settings can significantly boost performance for many common workloads, but as with any optimization, it's important to test thoroughly in your specific environment.
+
+## system variables
```

Comment:
```suggestion
## System variables
```
Please use sentence case (capitalize only the first word and proper nouns) for all headings at H2 level and below.